compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -1594,6 +1594,72 @@ TR::Register *OMR::Power::TreeEvaluator::vstoreEvaluator(TR::Node *node, TR::Cod
    return NULL;
    }
 
+TR::Register *OMR::Power::TreeEvaluator::awrtbarEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   // The wrtbar IL op represents a store with side effects.
+   // Currently we don't use the side effect node. So just evaluate it and decrement the reference count.
+   TR::Node *sideEffectNode = node->getSecondChild();
+   cg->evaluate(sideEffectNode);
+   cg->decReferenceCount(sideEffectNode);
+   // Delegate the evaluation of the remaining children and the store operation to the storeEvaluator.
+   return TR::TreeEvaluator::lstoreEvaluator(node, cg);
```

Comment:
Why is it lstoreEvaluator and not astoreEvaluator?

Reply:
Gita, thanks for catching that! Changing to astoreEvaluator.